sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1307,6 +1311,15 @@ static String getExtractDestinationUri(String extractDestinationDir) {
       return toBuilder().setNumFileShards(numFileShards).build();
     }
 
+    public Write<T> withCustomGcsTempLocation(ValueProvider<String> customGcsTempLocation) {
+      return toBuilder().setCustomGcsTempLocation(customGcsTempLocation).build();
+    }
+
+    public Write<T> withCustomGcsTempLocation(String customGcsTempLocation) {
+      return toBuilder().setCustomGcsTempLocation(
```

Comment:
Nit: easier to say "return withCustomGcsTempLocation(StaticValueProvider.of())"

Reply:
done